lib/validator.js

```diff
@@ -209,13 +210,13 @@ async function validateId() {
 }
 
 async function validateOrigin() {
-    var expectedOrigin = this.expectations.get("origin");
+    var expectedRpId = this.expectations.get("rpId");
     var clientDataOrigin = this.clientData.get("origin");
 
     var origin = checkOrigin(clientDataOrigin);
 
-    if (origin !== expectedOrigin) {
-        throw new Error("clientData origin did not match expected origin");
+    if (!new URL(origin).hostname.endsWith(expectedRpId)) {
```

Comment:
This is the relaxation mentioned in the spec? https://www.w3.org/TR/webauthn/#relying-party-identifier